doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst

```diff
@@ -0,0 +1,883 @@
+.. image:: first_stages.png
+   :width: 700px
+
+If you haven't made your own package, but still want to see what this looks like, you can launch this file from the tutorials: ::
+
+    ros2 launch moveit2_tutorials mtc_demo_minimal.launch.py
+
+Adding Stages
+-------------
+
+So far, we've walked through creating and executing a simple task, which runs but does not do much. Now, we will start adding the pick-and-place stages to the task. The image below shows an outline of the stages we will use in our task. To understand more about the concepts behind MoveIt Task Constructor and the different stage types, see the :doc:`example page for MoveIt Task Constructor </doc/examples/moveit_task_constructor/moveit_task_constructor_tutorial>`.
+
+.. image:: stages.png
+   :width: 700px
+
+We will start adding stages after our existing open hand stage here:
+
+.. code-block:: c++
+
+  auto stage_open_hand = std::make_unique<moveit::task_constructor::stages::MoveTo>("open hand", interpolation_planner);
+  stage_open_hand->setGroup(hand_group_name);
+  stage_open_hand->setGoal("open");
+  task.add(std::move(stage_open_hand));
+
+  // Add the next lines of codes to define more stages here
+
+Pick Stages
+^^^^^^^^^^^
+
+We need to move the arm to a position where we can pick up our object. This is done with a ``Connect`` stage, which as its name implies, is a Connector stage. This means that it tries to bridge between the results of the stage before and after it. This stage is initialized with a name, ``move_to_pick``, and a ``GroupPlannerVector`` that specifies the planning group and the planner. We then set a timeout for the stage, set the properties for the stage, and add it to our task.
+
+.. code-block:: c++
+
+  auto stage_move_to_pick = std::make_unique<moveit::task_constructor::stages::Connect>(
+      "move to pick",
+      moveit::task_constructor::stages::Connect::GroupPlannerVector{ { arm_group_name, sampling_planner } });
+  stage_move_to_pick->setTimeout(5.0);
+  stage_move_to_pick->properties().configureInitFrom(moveit::task_constructor::Stage::PARENT);
+  task.add(std::move(stage_move_to_pick));
+
+Next, we create a pointer to a MoveIt Task Constructor stage object, and set it to ``nullptr`` for now. Later, we will use this to save a stage.
+
+.. code-block:: c++
+
+  moveit::task_constructor::Stage* attach_object_stage =
+      nullptr;  // Forward attach_object_stage to place pose generator
+
+This next block of code creates a ``SerialContainer``. This is a container that can be added to our task and can hold several substages. In this case, we create a serial container that will contain the stages relevant to the picking action. Instead of adding the stages to the task, we will add the relevant stages to the serial container.
+
+.. code-block:: c++
+
+  {
+    auto grasp = std::make_unique<moveit::task_constructor::SerialContainer>("pick object");
+    task.properties().exposeTo(grasp->properties(), { "eef", "hand", "group", "ik_frame" });
```

Comment:
You could mention here that `exposeTo` declares the task properties from the parent task in the new grasp container, `configureInitFrom()` initializes them. This is necessary for allowing the contained stages to access them